### DISCUSSION TITLE
Request for additional template variables

### DISCUSSION THREAD

**Discussion by Matthew-Mind** [gildas-lormeau/SingleFile#1410] [Ideas]
{month-locale} expressed as a three letter abbreviation outputting Mar instead of 3
{day-locale} but outputting the three letter abbreviation for the day of the week instead of the number
Using these two abbreviations plus {day-locale} I could output some variation of:
Sun 10 Mar

{hours-locale} expressed on a 12 hour timescale
This could then be paired with:
{AM/PM-locale} 
Allowing me to display some variation of:
6 32 PM

Or less customizable but probably just as good:
{time-locale} with seconds excluded to output: 6_32 PM

**Comment by gildas-lormeau** [comment]
Thank you for the suggestion. The problem with dates is that I'm always asked to do very specific things, and I feel like I'm going to create thousands of variables if I respond to requests. I think it would be better to expose template functions to make the `DateTimeFormat` API accessible to users, cf. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat

**Comment by Matthew-Mind** [comment]
This does sound like the best solution.  Please note that in its current state SingleFile already works very well, and I really appreciate the work you have put in creating and supporting it.  If you decide to make the API accessible I will appreciate that as well, but it is not required, and certainly not a priority.

**Comment by gildas-lormeau** [comment]
It's implemented because it's a recurrent issue in fact. The API is exposed in the version `1.22.53` via the new template function `%datetime-custom`.
